hydroflow_plus/src/properties.rs

```diff
@@ -0,0 +1,89 @@
+use std::collections::HashSet;
+
+use stageleft::*;
+
+use crate::ir::{HfPlusLeaf, HfPlusNode, SeenTees};
+
+// schema : table per algebraic property with the list of expressions that satisfy the property
+
+// interface: can "tag" an expression with a property and it will add it to that table
+// can also run a check to see if an expression satisfies a property
+#[derive(Default)]
+pub struct PropertyDatabase {
+    //
+    commutative: HashSet<syn::Expr>,
+}
+
+#[allow(dead_code)]
+// allows us to convert the hydroflow datatype for folds to a binary operation for the algebra property tests
```

Comment:
```rust
/// allows us to convert the hydroflow datatype for folds to a binary operation for the algebra property tests
#[allow(dead_code)]
```

Reply:
Where is this actually used?

Reply:
we don't use it yet, we put it in because we are going to need it when we put in algebraic property checks.